cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -837,6 +834,54 @@ private void setOutputsIfNeeded(Job job) {
     }
   }
 
+  private void fixOutputPermissions(Job job, Map<String, ProvidedOutput> outputs) {
+    Set<String> outputsWithUmask = new HashSet<>();
+    Set<String> outputUmasks = new HashSet<>();
+    for (Map.Entry<String, ProvidedOutput> entry : outputs.entrySet()) {
+      String umask = entry.getValue().getOutputFormatConfiguration().get("fs.permissions.umask-mode");
```

Comment:
I was hoping to find a constant in Hadoop but could not find one... will define one